### ISSUE TITLE
Non-blocking audio playback in GUI

### ISSUE THREAD

**Issue by luisArandas** [pyimgui/pyimgui#238]
Hello all. 

I am currently using `python sounddevice ` to generate audio data while the GUI serves as the interface with the backend. Is there any way to **not** freeze the GUI while playing audio?

Everytime I play a buffer the GUI freezes until it ends, so you can image that this blocks for 2 minutes if I loop it for 2 minutes. This happens even in a separate thread..

Any idea on how to solve this?

**Comment by learn-more** [comment]
This does not seem to be related to pyimgui at all?
Anything on your 'main' thread will 'freeze' while you are executing something else.

**Comment by luisArandas** [comment]
Yes, this is python-specific. I was just wondering if the library had some workaround on this, similar to the QThread. Or are you saying that, I guess thanks to the global interpreter lock, there is no way for me to play audio in the background? Even using multiprocessing?

**Comment by KinoxKlark** [comment]
I don't think there are any threading functionalities provided by pyimgui, but it is not really its purpose. Pyimgui stands for "immediate mode GUI" which means that you call when you want to display. The management of the windows and various threads isn't done via pyimgui but via the rendering backend (pygame, glfw, sdl2, etc). I think you should look there for doing what you want to do efficiently. You can also try to use a multithreading module to instantiate a thread for your audio playing process in parallel to the windows (I do not have any name in my head but you should be able to find that by googling a bit). It is certainly possible to play audio in the background but it is not a task for pyimgui.

Hope this will help you a bit, don't hesitate to ask if something is unclear (or to correct me if I'm wrong).

**Comment by luisArandas** [comment]
Thank you for the clarification @KinoxKlark!
